test/ptp.go

```diff
@@ -0,0 +1,425 @@
+					Expect(len(ptpSupportedInt)).To(Equal(len(ptpDiscoveredInterfaces)), fmt.Sprint("The interfaces discovered incorrectly"))
+					for _, intfc := range ptpSupportedInt {
+						Expect(ptpDiscoveredInterfaces).To(ContainElement(intfc))
+					}
+				}
+			})
+
+			// 25730
+			It("The virtual interfaces should be not discovered by ptp", func() {
+
+				for _, pod := range ptpRunningPods {
+					ptpNotSupportedInt := getNonPtpMasterSlaveAttachedInterfaces(pod)
+					ptpDiscoveredInterfaces := ptpDiscoveredInterfaceList(nodePtpDeviceAPIPath + pod.Spec.NodeName)
+					for _, inter := range ptpNotSupportedInt {
+						Expect(ptpDiscoveredInterfaces).ToNot(ContainElement(inter), fmt.Sprint("The interfaces discovered incorrectly. PTP non supported Interfaces in list"))
+					}
+				}
+			})
+
+			// 25733
+			It("PTP daemon apply match rule based on nodeLabel", func() {
+				profileSlave := "Profile Name: slave"
+				profileMaster := "Profile Name: grandmaster"
+				for _, pod := range ptpRunningPods {
+					podLogs, err := pods.GetLog(&pod)
+					Expect(err).NotTo(HaveOccurred(), "Error to find needed log due to %s", err)
+					if podRole(pod, ptpSlaveNodeLabel) {
+						Expect(podLogs).Should(ContainSubstring(profileSlave),
+							fmt.Sprintf("Profile \"%s\" not found in pod's log %s", profileSlave, pod.Name))
+					}
+					if podRole(pod, ptpGrandmasterNodeLabel) {
+						Expect(podLogs).Should(ContainSubstring(profileMaster),
+							fmt.Sprintf("Profile \"%s\" not found in pod's log %s", profileSlave, pod.Name))
+					}
+				}
+			})
+
+			// 25738
+			It("Slave can sync to master", func() {
+				var masterID string
+				var slaveMasterID string
+				grandMaster := "assuming the grand master role"
+				for _, pod := range ptpRunningPods {
+					if podRole(pod, ptpGrandmasterNodeLabel) {
+						podLogs, err := pods.GetLog(&pod)
+						Expect(err).NotTo(HaveOccurred(), "Error to find needed log due to %s", err)
+						Expect(podLogs).Should(ContainSubstring(grandMaster),
+							fmt.Sprintf("Log message \"%s\" not found in pod's log %s", grandMaster, pod.Name))
+						for _, line := range strings.Split(podLogs, "\n") {
+							if strings.Contains(line, "selected local clock") && strings.Contains(line, "as best master") {
+								masterID = strings.Split(line, " ")[4]
+							}
+						}
+					}
+					if podRole(pod, ptpSlaveNodeLabel) {
+						podLogs, err := pods.GetLog(&pod)
+						Expect(err).NotTo(HaveOccurred(), "Error to find needed log due to %s", err)
+						for _, line := range strings.Split(podLogs, "\n") {
+							if strings.Contains(line, "new foreign master") {
+								slaveMasterID = strings.Split(line, " ")[6]
+							}
+						}
+					}
+				}
+				Expect(masterID).NotTo(BeNil())
+				Expect(slaveMasterID).NotTo(BeNil())
+				Expect(slaveMasterID).Should(HavePrefix(masterID), "Error match MasterID with the SlaveID. Slave connected to another Master")
+			})
+		})
+	})
+
+	Describe("Test NodeName Selector", func() {
```

Comment:
It looks to me this is testing if high priority profile can be applied?

Reply:
I change the name in this test we test both the NodeName selector and the priority function